src/object-store/src/util.rs

```diff
@@ -123,6 +123,24 @@ pub fn normalize_path(path: &str) -> String {
     p
 }
 
+// This logical tries to extract parent path from the object storage operation
+// the function also relies on assumption that the region path is built from
+// pattern `<data|index>/catalog/schema/table_id/....`
+//
+// this implementation is also safe when the path is not constructed in a
+// way that we assumed.
+pub fn extract_parent_path(path: &str) -> String {
```

Comment:
We can support the path level like the official layer.
https://docs.rs/opendal/latest/opendal/layers/struct.PrometheusLayer.html#method.enable_path_label

Reply:
Just changed my mind... I changed this implementation to return full path when there is no more than 3 levels. But I'm not going to make it fully configurable for levels...

Because this scenario is highly associated our path naming pattern. Having a constant number 3 for each setup can be more confusing than this centralized one.

Reply:
This implementation should be more memory efficient, could we port it?
https://github.com/apache/opendal/blob/174bda53f79123cd114d2409189423a0a4cf6bf3/core/src/layers/prometheus.rs#L880-L889

Reply:
Yes this one is way better. It's simply substring. I lost my mind when seeking the correct root path.